core/src/subgraph/registrar.rs

```diff
@@ -466,18 +466,26 @@ fn create_subgraph(
     Ok(CreateSubgraphResult { id: entity_id })
 }
 
-/// Resolves the chain head block and the subgraph's earliest block
+/// Resolves the chain head block, the subgraph's earliest block, and
+/// the manifest's graft base block
 fn resolve_subgraph_chain_blocks(
     manifest: SubgraphManifest,
     chain_store: Arc<impl ChainStore>,
     ethereum_adapter: Arc<dyn EthereumAdapter>,
     logger: &Logger,
 ) -> Box<
     dyn Future<
-            Item = (Option<EthereumBlockPointer>, Option<EthereumBlockPointer>),
+            Item = (
+                Option<EthereumBlockPointer>,
+                Option<EthereumBlockPointer>,
+                Option<EthereumBlockPointer>,
+            ),
```

Comment:
We should introduce a struct for this.

Reply:
I thought about that, but the function is used only once, in `create_subgraph_version`, and really is just a detail of that function. We should just rewrite `resolve_subgraph_chain_blocks` as three async functions and then `await` them in `create_subgraph_version`.

Reply:
I wasn't planning on doing that in this PR ;)